### DISCUSSION TITLE
"Enrolling" Amazfit watch with MQTT Explorer

### DISCUSSION THREAD

**Discussion by SteveDockar** [ESPresense/ESPresense#1202] [Q&A] [answered]
Not so much a question as an answer to the question "how do you enroll something like a device with a static BLE MAC that can't enroll through the UI?"

When I enable "Discoverable" in the Amazfit/Zepp app, I can see my watch in ESPresense /ui/#/fingerprints as `mifit:<device mac>`. The BLE MAC doesn't rotate so I can use this as the identifier in Home Assistant which is great. The default RSSI@1m is too low for the watch and consequently, the distance reads too close. Unlike my Samsung Galaxy phone, I can't enroll the watch through the ESPresense UI because the watch doesn't support pairing, except with a phone and "headphones". I couldn't find a solution documented in one place, but the answer is scattered around here and on the HA forums _if_ you know what you're looking for. So, I decided to write it up here in case anyone else is looking to solve a similar issue.

Here's the watch in MQTT Explorer in the espresense/devices/ topic:
![image](https://github.com/ESPresense/ESPresense/assets/22272895/3ddd33c2-f824-4946-9a54-d08d54108e5e)

And to "Enroll" it, I copied the message format of an already enrolled device from the espresense/settings topic: 
![image](https://github.com/ESPresense/ESPresense/assets/22272895/59e47df8-4ac2-4d34-a552-34062ca21a55)
This is my enrolled phone which rotates it's MAC every so often, so is stored in settings under it's Identity Resolving Key (IRK). Notice how the details are in the _device's_ /config "sub-topic".

And in MQTT Explorer I took the mifit:\<device mac\> ID from the /devices/ topic and used it to create a new "_device_" under the /settings/ topic, using the /settings/mifit:\<device mac\>/config sub-topic. There I published a message with the JSON object I copied from my phone's config message but with the watch  ID, name and signal level I wanted:
![image](https://github.com/ESPresense/ESPresense/assets/22272895/c728fe19-30a4-44c5-b193-7d56e3e98fb2)

To ensure this survives reboots etc. make sure to set the "Retain" flag before you publish
![image](https://github.com/ESPresense/ESPresense/assets/22272895/b54fd2de-8b48-4809-9088-92e6a26708cf)
 (or if you forgot, just publish the same message to the same topic but _with_ the Retain flag set this time. 🤦 )

If your device rotates its BLE MAC address periodically and you see it repeated in the fingerprints list then this approach won't work I'm afraid. I was lucky that the device I wanted to track couldn't enroll through the UI, but did keep the same MAC so I could use that as its identifier.

I hope somebody finds this useful.

**Comment by Terastar-Paperclip** [comment, marked as answer]
This is a "works, with quirks" device per our enrollment reference (#2324) — MQTT Explorer + manual named enrollment is the right path; the docs walkthrough is linked from the matrix there. Marking this answered off that pin — shout if the Amazfit-specific steps don't match what you're seeing and we'll fix the docs.
